modules/pulse/pulse_sim.py

```diff
@@ -1,181 +1,234 @@
-            if NAMES.TRIG in changes:
-                trig_rise = (self.TRIG_EDGE == 0 and self.TRIG) or \
-                            (self.TRIG_EDGE == 1 and not self.TRIG) or \
-                            self.TRIG_EDGE == 2
-                trig_fall = not trig_rise
-            else:
-                trig_rise = False
-                trig_fall = False
-            queue_full = len(self.queue) + 2 > MAX_QUEUE
-
-            # If we got a pulse that won't fit, error
-            if trig_rise and (ts < self.valid_ts or queue_full):
-                self.DROPPED += 1
-            elif trig_rise:
-                if delay == 0:
-                    # Produce pulse now
-                    out = 1
-                else:
-                    # Queue the rising edge
-                    self.do_queue(ts + delay, 1)
-                # Store that we are currently outputting a pulse and when it
-                # started
-                self.ongoing_pulse = 1
-                self.rising_ts = ts
-                # We will need to be called next clock tick to queue the
-                # falling edge
-                if width != 0:
-                    next_ts = ts + 1
-            elif (width == 0 and trig_fall and self.ongoing_pulse) or \
-                    (width != 0 and self.ongoing_pulse):
-                # This is when we need to do something with the falling edge
-                if width == 0:
-                    self.produced_width = ts - self.rising_ts
-                else:
-                    self.produced_width = width
-                if delay == 0 and width == 0:
-                    # Bypass the queue, produce it straight away
-                    out = 0
-                else:
-                    # Queue the falling edge
-                    self.do_queue(
-                        self.rising_ts + delay + self.produced_width, 0)
-                # We have made the first rising and falling edge, so queue the
-                # rest
-                self.edges_remaining = 2 * pulses - 2
-                # Say that we have stopped outputting a pulse
-                self.ongoing_pulse = 0
-                # Can't get another pulse until we produced the falling edge
-                self.valid_ts = self.rising_ts + (pulses - 1) * step \
-                    + self.produced_width + MIN_QUEUE_DELTA
-
-            # If we now have edges_remaining, or if we didn't update the queue
-            # yet then we need to process then next clock tick
-            if len(self.queue) != self.QUEUED:
-                next_ts = ts + 1
-
-            # Set the output, which might have been set from the queue, or
-            # might be produced straight through
-            self.OUT = out
-
-        if next_ts:
-            assert next_ts >= ts, "Going back in time %s >= %s" % (
-                next_ts, ts)
-            return next_ts
+	ENABLE, TRIG, DELAY_L, DELAY_H, WIDTH_L, WIDTH_H, PULSES, \
```

Comment:
It looks to me like this file is now formatted with hard tabs.  I'm sure this is not in keeping with the project style.

Reply:
Sorted.